fe/fe-core/src/main/java/com/starrocks/memory/MemoryUsageTracker.java

```diff
@@ -0,0 +1,135 @@
+        registerMemoryTracker("Load", currentState.getRoutineLoadMgr());
+        registerMemoryTracker("Load", currentState.getStreamLoadMgr());
+
+        registerMemoryTracker("Export", currentState.getExportMgr());
+        registerMemoryTracker("Delete", currentState.getDeleteMgr());
+        registerMemoryTracker("Transaction", currentState.getGlobalTransactionMgr());
+        registerMemoryTracker("Backup", currentState.getBackupHandler());
+        registerMemoryTracker("Task", currentState.getTaskManager());
+        registerMemoryTracker("Task", currentState.getTaskManager().getTaskRunManager());
+        registerMemoryTracker("Tablet", currentState.getTabletInvertedIndex());
+        registerMemoryTracker("Profile", ProfileManager.getInstance());
+        registerMemoryTracker("LocalCatalog", new InternalCatalogMemoryTracker());
+
+        QeProcessor qeProcessor = QeProcessorImpl.INSTANCE;
+        if (qeProcessor instanceof QeProcessorImpl) {
+            registerMemoryTracker("Coordinator", (QeProcessorImpl) qeProcessor);
+        }
+
+        IDictManager dictManager = IDictManager.getInstance();
+        if (dictManager instanceof CacheDictManager) {
+            registerMemoryTracker("Dict", (CacheDictManager) dictManager);
+        }
+
+        LOG.info("Memory usage tracker init success");
+
+        initialize = true;
+    }
+
+    public static void registerMemoryTracker(String moduleName, MemoryTrackable object) {
+        REFERENCE.computeIfAbsent(moduleName, k -> new HashMap<>());
+        REFERENCE.get(moduleName).put(object.getClass().getSimpleName(), object);
+    }
+
+    public static void trackMemory() {
+        long startTime;
+        long endTime;
+        for (Map.Entry<String, Map<String, MemoryTrackable>> entry : REFERENCE.entrySet()) {
+            String moduleName = entry.getKey();
+            Map<String, MemoryTrackable> statMap = entry.getValue();
+            MemoryStat memoryStat = new MemoryStat();
+            long estimateSize = 0L;
+            long estimateCount = 0L;
+            for (Map.Entry<String, MemoryTrackable> statEntry : statMap.entrySet()) {
+                String className = statEntry.getKey();
+                MemoryTrackable tracker = statEntry.getValue();
+                startTime = System.currentTimeMillis();
+                long currentEstimateSize = tracker.estimateSize();
+                Map<String, Long> counterMap = tracker.estimateCount();
+                endTime = System.currentTimeMillis();
+                estimateSize += currentEstimateSize;
+
+                StringBuilder sb  = new StringBuilder();
+                for (Map.Entry<String, Long> subEntry : counterMap.entrySet()) {
+                    sb.append(subEntry.getKey()).append(" with ").append(subEntry.getValue())
+                            .append(" object(s). ");
+                }
+
+                LOG.info("({}ms) Module {} - {} estimated {} of memory. Contains {}",
```

Comment:
please add a unit for reader friendly?

Reply:
There are units, which have been converted in the code.